### PR TITLE
Bug fixes to delete persons endpoint

### PR DESCRIPTION
**Pull request by sher812**
Fixed bugs and re-tested with postman:

Testcases (When user wants to delete persons linked with):
[x] no encounter, single person
[x] Single Encounter, Single Person
[x] Single Encounter, Multiple Persons
[x] Multiple Encounters, Single Person
[x] Multiple Encounters, Multiple Persons

Fix(#184)

### CODE REVIEW

File: backend/src/services/user.service.ts [hunk 1]
```diff
@@ -25,16 +25,26 @@ export const getUserByAuthId = async (uid) => {
...
   } else {
     return false;
   }
-};
+}
+
+export const deleteUserEncounter = async (encounterID: String) => {
+  const result = await User.updateMany({ }, { $pullAll: {encounters: [{ _id: encounterID}]}});
+
+  if (result.modifiedCount > 0) {
```
Comment (schi314) [inline backend/src/services/user.service.ts:41]: Is the relationship between User and Encounter one-to-many? If so then an encounter would only ever belong to one User, and `result.modifiedCount == 1` would be more precise?

Comment (sher812) [reply backend/src/services/user.service.ts:41]: Ah yes will change that

**Review by butterplup — APPROVED** [review]
LG TM 🐸

**Review by UoA-sjer215 — APPROVED** [review]
lgtm

**Pull request merged**